ddtrace/llmobs/_llmobs.py

```diff
@@ -1389,69 +1398,64 @@ def submit_evaluation(
         """
         Submits a custom evaluation metric for a given span ID and trace ID.
 
-        :param span_context: A dictionary containing the span_id and trace_id of interest.
+        :param dict span_context: A dictionary of shape {'span_id': str, 'trace_id': str} uniquely identifying
+                            the span associated with this evaluation.
         :param str label: The name of the evaluation metric.
-        :param str metric_type: The type of the evaluation metric. One of "categorical", "score".
+        :param str metric_type: The type of the evaluation metric. One of "categorical", "score", "boolean".
         :param value: The value of the evaluation metric.
-                      Must be a string (categorical), integer (score), or float (score).
+                      Must be a string (categorical), integer (score), float (score), or boolean (boolean).
         :param tags: A dictionary of string key-value pairs to tag the evaluation metric with.
         :param str ml_app: The name of the ML application
-        :param int timestamp_ms: The timestamp in milliseconds when the evaluation metric result was generated.
+        :param int timestamp_ms: The unix timestamp in milliseconds when the evaluation metric result was generated.
+                                 If not set, the current time will be used.
         :param dict metadata: A JSON serializable dictionary of key-value metadata pairs relevant to the
                                 evaluation metric.
         """
         if cls.enabled is False:
             log.debug(
-                "LLMObs.submit_evaluation() called when LLMObs is not enabled. Evaluation metric data will not be sent."
+                "LLMObs.submit_evaluation() called when LLMObs is not enabled. "
+                "Evaluation metric data will not be sent."
             )
             return
+
+        ml_app = ml_app if ml_app else config._llmobs_ml_app
+        if not ml_app:
+            log.warning(
+                "ML App name is required for sending evaluation metrics. Evaluation metric data will not be sent. "
+                "Ensure this configuration is set before running your application."
+            )
+            telemetry.record_llmobs_submit_evaluation({"span": span_context}, metric_type, "missing_ml_app")
+            return
```

Comment:
Let's avoid moving this block of code around, I don't see the benefit in moving this outside of the try/catch